docs/api-spec.md

```diff
@@ -0,0 +1,322 @@
+    + [Metadata](#metadata)
+    + [Spec](#spec)
+    + [Status](#status)
+- [Status Signalling](#status-signalling)
+- [Listing Resources](#listing-resources)
+- [Detailed Resource Types - v1beta1](#detailed-resource-types---v1beta1)
+  * [`ArrayOrString`](#arrayorstring)
+  * [`ContainerStateRunning`](#containerstaterunning)
+  * [`ContainerStateWaiting`](#containerstatewaiting)
+  * [`ContainerStateTerminated`](#containerstateterminated)
+  * [`EnvVar`](#envvar)
+  * [`Param`](#param)
+  * [`ParamSpec`](#paramspec)
+  * [`Step`](#step)
+  * [`StepState`](#stepstate)
+  * [`TaskResult`](#taskresult)
+  * [`TaskRunResult`](#taskrunresult)
+  * [`TaskSpec`](#taskspec)
+  * [`WorkspaceBinding`](#workspacebinding)
+  * [`WorkspaceDeclaration`](#workspacedeclaration)
+<!-- /toc -->
+
+## Abstract
+
+The Tekton Pipelines platform provides common abstractions for describing and executing container-based, run-to-completion workflows, typipcally in service of CI/CD scenarios. This document describes the structure, lifecycle and management of Knative resources in the context of the [Kubernetes Resource Model](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/resource-management.md). An understanding of the Kubernetes API interface and the capabilities of [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) is assumed.
+
+This document does not define the [runtime contract](https://tekton.dev/docs/pipelines/container-contract/) nor prescribe specific implementations of supporting services such as access control, observability, or resource management.
+
+This document makes reference in a few places to different profiles for Tekton installations. A profile in this context is a set of operations, resources, and fields that are accessible to a developer interacting with a Tekton installation. Currently, only a single (minimal) profile for Tekton Pipelines is defined, but additional profiles may be defined in the future to standardize advanced functionality. A minimal profile is one that implements all of the “MUST”, “MUST NOT”, and “REQUIRED” conditions of this document.
+
+## Background
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+There is no formal specification of the Kubernetes API and Resource Model. This document assumes Kubernetes 1.13 behavior; this behavior will typically be supported by many future Kubernetes versions. Additionally, this document may reference specific core Kubernetes resources; these references may be illustrative (i.e. an implementation on Kubernetes) or descriptive (i.e. this Kubernetes resource MUST be exposed). References to these core Kubernetes resources will be annotated as either illustrative or descriptive.
```

Comment:
Oh probably just because the original draft was written when 1.13 was the minimum required. 👴 

AFAIK no relevant fields have been removed/changed since 1.13, so it's technically still true. 🤷